pkg/auth/auth.go

```diff
@@ -53,6 +54,61 @@ func AuthBackend(shard conn.DBInstance, berule *config.BackendRule, msg pgproto3
 			return fmt.Errorf("no auth rule specified for server connection")
 		}
 		return shard.Send(&pgproto3.PasswordMessage{Password: berule.AuthRule.Password})
```

Comment:
what about authFrontend() ? 

also, we can actually setup postgresql shard with scram auth in tests and run scram auth in CI....